### PR TITLE
front: disable electrical profile option for thermal rollingstock

### LINKED ISSUE #4909: front: change the message for thermal mode in the linear electrical profile view
## Description
In case of thermal rolling stock, in the linear profile the message "mode incompatible" is not correct :
Image
For thermal rolling stock any information about elecrification should be displayed. ATTENTION : some rolling stock could be double mode (thermal/1500V or thermal/25000 V). In that cases, info about electrical profile should be displayed on the path only in the zones where the rolling stock uses electrical traction.
 
## AC
Information under the GEV (checkbox "electrical profile" activated) should be changed in the following cases:
- [ ] If the RS can run only in thermal mode, the button to display the "electrical profile" is not active
- [ ] If "electrical profile" was active and we switch to a train with a RS who can run only in thermal mode, the checkbox should be unchecked and the option disabled
- [ ] If the RS is bi-mode and it runs in thermal mode on the whole path, the button to display the "electrical profile" is not active
- [ ] If the RS is bi-mode and it runs in electrical mode on a specfic zone, information about electrification mode and electrical profiles should be displayed in that zone

### LINKED ISSUE #4948: front : electrical profiles options in GEV not readable
### What happened?

When the text displayed in the GEV is too long, it goes beyond the limit of the background and the text is not properly readable

image

### What did you expect to happen?

The background should scale with the text content

### How can we reproduce it (as minimally and precisely as possible)?

1. Go to a scenario
2. Add a train with an incompatible electrical profile on the path
3. On the GEV, activate the "Electrical Profiles" option

### What operating system, browser and environment are you using?

- **Browser**: Firefox v117
- **OS**: Ubuntu
- **Env**: Dev

### OSRD version (top right corner `Account` button > `Informations`)

01d09d94

### PR DESCRIPTION
**Pull request by SharglutDev**
close #4909 
close #4948

### CODE REVIEW

File: front/src/common/BootstrapSNCF/CheckboxRadioSNCF.tsx
Comment (clarani) [inline front/src/common/BootstrapSNCF/CheckboxRadioSNCF.tsx:1]: Nice, thanks for migrating this file to ts :)
And the refacto is very very nice :tada:

File: front/src/common/BootstrapSNCF/CheckboxRadioSNCF.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  disabled?: boolean;
+};
+
+const CheckboxRadioSNCF = ({
+  id,
+  label,
+  name,
+  checked = false,
+  onChange,
+  type = 'checkbox',
```
Comment (clarani) [inline front/src/common/BootstrapSNCF/CheckboxRadioSNCF.tsx]: No need for these values by default :)
```suggestion
  checked,
  onChange,
  type,
```

File: front/src/common/BootstrapSNCF/CheckboxRadioSNCF.scss
Comment (clarani) [inline front/src/common/BootstrapSNCF/CheckboxRadioSNCF.scss:1]: I think I need @nicolaswurtz's point of vue on this :)

File: front/src/applications/operationalStudies/components/SimulationResults/ChartHelpers/ChartHelpers.ts [hunk 2]
```diff
@@ -364,3 +364,20 @@ export function makeTrainList(trains: Train[], trainToOffset: number, dragOffset
 export function makeTrainListWithAllTrainsOffset(trains: Train[], dragOffset = 0) {
   return trains.map((train) => trainWithDepartureAndArrivalTimes(train, dragOffset));
 }
+
+/**
+ * Get the width of an element based of it's text content
```
Comment (clarani) [inline front/src/applications/operationalStudies/components/SimulationResults/ChartHelpers/ChartHelpers.ts]:
```suggestion
 * Get the width of an element based on its text content
```

File: front/src/applications/operationalStudies/components/SimulationResults/ChartHelpers/drawElectricalProfile.jsx [hunk 3]
```diff
@@ -74,12 +75,15 @@ const drawElectricalProfile = (
     if ((rotate && height < -24) || (!rotate && width > 60)) {
       const textZone = chart.drawZone.select(`#${groupID}`);
 
+      // get the width of the text element and adding a few pixels for readability
+      const svgWidth = getElementWidth(dataSimulation.text, 8, `#${groupID}`) + 10;
```
Comment (clarani) [inline front/src/applications/operationalStudies/components/SimulationResults/ChartHelpers/drawElectricalProfile.jsx]: Add a constant here to improve the readability
```suggestion
      const labelFontSize = 8; // in px
      const svgWidth = getElementWidth(dataSimulation.text, labelFontSize, `#${groupID}`) + 10;
```

**Review by clarani — COMMENTED** [review]
Tested in local :heavy_check_mark: 
Thanks for this PR, very nice job ! :)

File: front/src/common/BootstrapSNCF/CheckboxRadioSNCF.tsx (hunk 1 above)
Comment (SharglutDev) [reply front/src/common/BootstrapSNCF/CheckboxRadioSNCF.tsx]: done :)

File: front/src/applications/operationalStudies/components/SimulationResults/ChartHelpers/ChartHelpers.ts (hunk 2 above)
Comment (SharglutDev) [reply front/src/applications/operationalStudies/components/SimulationResults/ChartHelpers/ChartHelpers.ts]: done :)

File: front/src/applications/operationalStudies/components/SimulationResults/ChartHelpers/drawElectricalProfile.jsx (hunk 3 above)
Comment (SharglutDev) [reply front/src/applications/operationalStudies/components/SimulationResults/ChartHelpers/drawElectricalProfile.jsx]: done :)

File: front/src/common/BootstrapSNCF/CheckboxRadioSNCF.scss
Comment (SharglutDev) [reply front/src/common/BootstrapSNCF/CheckboxRadioSNCF.scss:1]: Had a meeting with @nicolaswurtz. He said, he's ok with the css like that. He just asked to use css variables for the color values which I did :)

**Review by clarani — APPROVED** [review]
LGTM :)

**Pull request merged**